jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageReader.java

```diff
@@ -62,6 +72,73 @@
     }
   }
 
+  /**
+   * Retrieves the cached manifest for an image reference.
+   *
+   * @param imageReference the image reference
+   * @return the manifest for the image reference, if found
+   * @throws IOException if an I/O exception occurs
+   */
+  Optional<ManifestTemplate> retrieveManifest(ImageReference imageReference) throws IOException {
+    Path imageDirectory = cacheStorageFiles.getImageDirectory(imageReference);
+    Path manifestPath = imageDirectory.resolve("manifest.json");
+    if (!Files.exists(manifestPath)) {
+      return Optional.empty();
+    }
+
+    try (LockFile ignored = LockFile.lock(imageDirectory.resolve("lock"))) {
+      // TODO: Consolidate with ManifestPuller
+      ObjectNode node =
```

Comment:
I feel like we have some code like this somewhere? How easy is it to share that logic? In `PullBaseImageStep` or something?

Reply:
There's very similar code in `ManifestPuller`, and I wasn't immediately sure how to share it, hence the todo. I wonder if we could stick it in `JsonTemplateMapper`, or even a static method in `ManifestTemplate` or something?